crates/gen/src/type_namespaces.rs

```diff
@@ -16,11 +16,20 @@ impl TypeNamespaces {
         self.0.par_iter().map(|(name, tree)| {
             let name = case::to_snake(name, MethodKind::Normal);
             let name = format_ident(&name);
-            let tree = tree.to_tokens().collect::<Vec<_>>();
+            let tokens = tree.to_tokens().collect::<Vec<_>>();
 
-            quote! {
-                pub mod #name {
-                    #(#tree)*
+            if tree.foundation {
+                quote! {
+                    pub mod #name {
+                        #(#tokens)*
+                        pub use ::winrt::foundation::*;
```

Comment:
```rust 
let foundation  = if tree.foundation { quote! { pub use ::winrt::foundation::* } else { TokenStream::new() };
 quote! {
     pub mod #name {
         #(#tokens)*
         #foundation 
     }
 }


```

Reply:
Also I thought we wanted to export foundation as is (to mirror the same hierarchry). So this should be `pub use::winrt::foundation` without the `*`.

Reply:
There's one foundation sub-namespace that's not pulled into winrt but I don't suppose there's much value in excluding it at this point.